smarts/core/smarts.py

```diff
@@ -205,35 +214,41 @@ def _step(self, agent_actions):
         self._vehicle_states = [v.state for v in self._vehicle_index.vehicles]
 
         # Agents
-        self._agent_manager.step_sensors(self)
+        with timeit("Stepping through sensors", self._log):
+            self._agent_manager.step_sensors(self)
 
         if self._renderer:
             # runs through the render pipeline (for camera-based sensors)
             # MUST perform this after step_sensors() above, and before observe() below,
             # so that all updates are ready before rendering happens per frame
-            self._renderer.render()
-
-        observations, rewards, scores, dones = self._agent_manager.observe(self)
+            with timeit("Running through render pipeline", self._log):
+                self._renderer.render()
 
-        response_for_ego = self._agent_manager.filter_response_for_ego(
-            (observations, rewards, scores, dones)
-        )
+        with timeit("calculating observations and rewards", self._log):
+            observations, rewards, scores, dones = self._agent_manager.observe(self)
+            response_for_ego = self._agent_manager.filter_response_for_ego(
+                (observations, rewards, scores, dones)
+            )
```

Comment:
nit: This is not important but I could suggest that the returned values sit on the outer level.
```suggestion
        observations, rewards, scores, dones = None, None, None, None
        response_for_ego = None
        with timeit("calculating observations and rewards", self._log):
            observations, rewards, scores, dones = self._agent_manager.observe(self)
            response_for_ego = self._agent_manager.filter_response_for_ego(
                (observations, rewards, scores, dones)
            )
```